experiments/nlp_experiments/classification/bert_mcdropout.py

```diff
@@ -0,0 +1,180 @@
+#                                     labels=target)[:2]
+#
+#         loss.backward()
+#         optimizer.step()
+#         self._update_metrics(text_fea, target, loss, filter='train')
+#         return loss
+#
+#     def test_on_batch(
+#             self,
+#             data: dict(),
+#             target: torch.Tensor,
+#             cuda: bool = False,
+#             average_predictions: int = 1
+#     ):
+#         with torch.no_grad():
+#             input_ids, attention_mask = data['input_ids'], data['attention_mask']
+#             if cuda:
+#                 input_ids, attention_mask, target = to_cuda(input_ids), to_cuda(attention_mask),\
+#                                                     to_cuda(target)
+#             loss, text_fea = self.model(input_ids=input_ids, attention_mask=attention_mask,
+#                                         labels=target)[:2]
+#             self._update_metrics(text_fea, target, loss, 'test')
+#             return loss
+#
+#     def predict_on_batch(self, data, iterations=1, cuda=False):
+#
+#         # TODO: do we need replicate in memory = False for NLP?
+#         with torch.no_grad():
+#             input_ids, attention_mask = data['input_ids'], data['attention_mask']
+#             if cuda:
+#                 input_ids, attention_mask = to_cuda(input_ids), to_cuda(attention_mask)
+#             if self.replicate_in_memory:
+#                 input_ids = map_on_tensor(lambda d: stack_in_memory(d, iterations), input_ids)
+#                 attention_mask = map_on_tensor(lambda d: stack_in_memory(d, iterations),
+#                                                attention_mask)
+#                 try:
+#                     out = self.model(input_ids=input_ids, attention_mask=attention_mask)[0]
+#                 except RuntimeError as e:
+#                     raise RuntimeError(
+#                         '''CUDA ran out of memory while BaaL tried to replicate data.
+#                         See the exception above. Use `replicate_in_memory=False` in order
+#                         to reduce the memory requirements.
+#                         Note that there will be some speed trade-offs''') from e
+#                 out = map_on_tensor(lambda o: o.view([iterations, -1, *o.size()[1:]]), out)
+#                 out = map_on_tensor(lambda o: o.permute(1, 2, *range(3, o.ndimension()), 0), out)
+#             return out
+
+
+def main():
+    args = parse_args()
+    use_cuda = torch.cuda.is_available()
+    torch.backends.cudnn.benchmark = True
+    random.seed(1337)
+    torch.manual_seed(1337)
+    if not use_cuda:
+        print("warning, the experiments would take ages to run on cpu")
+
+    hyperparams = vars(args)
+
+    active_set, test_set = get_datasets(hyperparams['initial_pool'])
+
+    heuristic = get_heuristic(hyperparams['heuristic'],
+                              hyperparams['shuffle_prop'])
+
+    model = BertForSequenceClassification.from_pretrained(
```

Comment:
We are missing MCDropoutModule no?

Reply:
good catch, yes, I havent test it yet that's why. but I wanted to make sure the design is good.